pandas/tests/series/methods/test_replace.py

```diff
@@ -208,6 +208,13 @@ def test_replace_with_dict_with_bool_keys(self):
         expected = pd.Series(["yes", False, "yes"])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("dtype", ["int8", "int16", "int32", "int64"])
+    def test_replace_int_with_na(self, dtype):
+        # GH 38267
+        result = pd.Series([0, None]).astype(dtype).replace(0, pd.NA)
+        expected = pd.Series([0, None]).astype(dtype).fillna(0).replace(0, pd.NA)
```

Comment:
Is it possible to construct `expected` directly (e.g. `expected = pd.Series([pd.NA, pd.NA])`)?